code/modules/reagents/chemistry/holder.dm

```diff
@@ -332,16 +343,50 @@
 		R.on_update (A)
 	update_total()
 
-/datum/reagents/proc/handle_reactions()
+//beaker check proc,
+/datum/reagents/proc/beaker_check(atom/A)
+	if(istype(A, /obj/item/reagent_containers/glass/beaker/meta) || istype(A, /obj/item/reagent_containers/glass/bottle)) //prevent bottles from being dispenced and melting.
```

Comment:
A much better way to handle this entire proc would be to pass this onto the reagent_container itself, and have them override the beaker check on their own as necessary, instead of all of the logic being hardcoded into the beaker_check() proc